packages/calcite-components/src/components/accordion/accordion.tsx

```diff
@@ -62,13 +81,22 @@ export class Accordion {
   //
   //--------------------------------------------------------------------------
 
+  connectedCallback(): void {
+    this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
```

Comment:
I _think_ `childList` should suffice since we would only update items when the direct children change, and not the children of each item (i.e., nesting is not supported).